homeassistant/components/glances/sensor.py

```diff
@@ -214,3 +217,7 @@ async def async_update(self):  # noqa: C901
                     self._state = round(mem_use / 1024 ** 2, 1)
             except KeyError:
                 self._state = STATE_UNAVAILABLE
+        elif self.entity_description.type == "raid":
+            for raid_device, raid in value["raid"].items():
+                if raid_device == self._sensor_name_prefix:
```

Comment:
Why don't we look up the correct key in the dict instead of iterating all the items of the dict?